### ISSUE TITLE
 Fatal error: Call to a member function getValueFormatter() on a non-object in "SMW_DV_String.php" on line 74

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticGlossary#26]
Originally reported [on mailing list](http://wikimedia.7.x6.nabble.com/SMW-upgrade-problem-tc5075651.html) by @krabina and copied from [phabricator](https://phabricator.wikimedia.org/T168398).

I tried to upgrade a test-wiki (MW 1.23) from SMW 2.4.x to 2.5.
upgrade.php works quite far, but then throws an error:
```
  Database initialized completed.
  Import of vocabulary.json ...
  ... creating MediaWiki:Smw import skos contents ...
  ... creating MediaWiki:Smw import foaf contents ...
  ... creating MediaWiki:Smw import owl contents ...
  ... creating Attribut:Foaf:knows contents ...
  ... creating Attribut:Foaf:name contents ...
  ... creating Attribut:Foaf:homepage contents ...
  ... creating Attribut:Owl:differentFrom contents ...
  ... done.
  Import processing completed.
  ...site_stats is populated...done.
  Fatal error: Call to a member function getValueFormatter() on a non-object in
  ../extensions/SemanticMediaWiki/includes/datavalues/SMW_DV_String.php on line 74
```
The solution was to deactivate Semantic Glossary. So this is probably some incompatibliity with Semantic Glossary and SMW 2.5?

**Comment by kghbln** [comment]
@krabina Could you confirm that you are using [version 1.1.2](https://github.com/SemanticMediaWiki/SemanticGlossary/blob/master/RELEASE-NOTES.md#200) for this.

**Comment by mwjames** [comment]
I just replied in the phap ticket on the reasoning.

**Comment by kghbln** [comment]
> I just replied in the phap ticket on the reasoning.

Basically you pointed at the same possible reason.

**Comment by krabina** [comment]
yes, I was using Semantic Glossary 1.x with composer

**Comment by kghbln** [comment]
> yes, I was using Semantic Glossary 1.x with composer

Since you are trying to use versions of SG and Lingo which should theoretically work it could be that you found an yet unknown incompatibility in connection with MW 1.23+ and SMW 2.5+. I am not sure if @s7eph4n will put effort into this since MW 1.23 is basically end of life now.

**Comment by mwjames** [comment]
> theoretically work it could be that you found an yet unknown incompatibility
> in connection with MW 1.23+ and SMW 2.5+. I am not sure if @s7eph4n will put

Just to clarify, 1.x is not compatible that is why we have 2.x and the
mentioned ticket.

There is not an unknown incompatibility because SG 1.x is no longer
maintained which means if someone installs it due to some requirement
(1.23 etc.) then this not an issue of the software maintainer.

... and as I recall the very email conversation, I mentioned earlier,
I replied with:

> > On MW 1.23 I have to use SG ~1.0, because ~2.0 requires MW 1.26

> Yes, I thought so, [0] addresses the 2.5 compat. You may request a
> back-port of [0] to the 1.x branch.

> [0] https://github.com/SemanticMediaWiki/SemanticGlossary/pull/20

**Comment by kghbln** [comment]
> Just to clarify, 1.x is not compatible that is why we have 2.x and the mentioned ticket.

Ah ok it too hot today (always these excuses ;). So to conclude SG 1.x and Lingo 1.x are not compatible with SMW 2.5+.

**Comment by mwjames** [comment]
> Closed #26.

For future reference:

A software that contains a major release comes with an assumption
(unless otherwise stated) that any other release with a lower major
release number is no longer maintained which means that any error or
issue by using those unmaintained releases is in the sole
responsibility of that user.

A user is free to modify or fork those non-maintained releases but
should not expect any support or maintenance activities from project
members.

While I personally can sympathize with user who see such policy as too
stringent and cannot easily switch releases due to environmental or
institutional limitations, time as scarce resource only allows for a
limited amount of attention to issues and maintenance activities and
therefore requires some form of protocol [0].

[0] http://semver.org/
